elide-core/src/main/java/com/yahoo/elide/core/EntityBinding.java

```diff
@@ -51,77 +50,63 @@
+    public final ConcurrentHashMap<String, Class<?>> fieldsToTypes = new ConcurrentHashMap<>();
+    public final ConcurrentHashMap<String, String> aliasesToFields = new ConcurrentHashMap<>();
+    public final ConcurrentHashMap<String, AccessibleObject> accessibleObject = new ConcurrentHashMap<>();
+
+    public final ConcurrentHashMap<Class<? extends Annotation>, Annotation> annotations = new ConcurrentHashMap<>();
 
     public static final EntityBinding EMPTY_BINDING = new EntityBinding();
 
     /* empty binding constructor */
     private EntityBinding() {
-        jsonApi = null;
+        jsonApiType = null;
         idField = null;
         idType = null;
-        attrsDeque = null;
-        attrs = null;
-        relationshipsDeque = null;
+        attributes = null;
         relationships = null;
-        relationshipTypes = null;
-        relationshipToInverse = null;
-        fieldsToValues = null;
-        fieldsToTypes = null;
-        fieldsToTriggers = new MultiValueMap();
-        aliasesToFields = null;
-        accessibleObject = null;
         entityClass = null;
-        annotations = null;
+        entityPermissions = EntityPermissions.EMPTY_PERMISSIONS;
     }
 
     public EntityBinding(Class<?> cls, String type) {
+        entityClass = cls;
+        jsonApiType = type;
+
         // Map id's, attributes, and relationships
         Collection<AccessibleObject> fieldOrMethodList = CollectionUtils.union(
                 Arrays.asList(cls.getFields()),
-                Arrays.asList(cls.getMethods()));
-
-        entityClass = cls;
-        jsonApi = type;
-        // Initialize our maps for this entity. Duplicates are checked above.
-        attrsDeque = new ConcurrentLinkedDeque<>();
-        relationshipsDeque = new ConcurrentLinkedDeque<>();
-        relationshipTypes = new ConcurrentHashMap<>();
-        relationshipToInverse = new ConcurrentHashMap<>();
-        fieldsToValues = new ConcurrentHashMap<>();
-        fieldsToTypes = new ConcurrentHashMap<>();
-        fieldsToTriggers = new MultiValueMap<>();
-        aliasesToFields = new ConcurrentHashMap<>();
-        accessibleObject = new ConcurrentHashMap<>();
-        annotations = new ConcurrentHashMap<>();
+                Arrays.asList(cls.getMethods())
+        );
         bindEntityFields(cls, type, fieldOrMethodList);
         bindAccessibleObjects(cls, fieldOrMethodList);
 
-        attrs = dequeToList(attrsDeque);
+        attributes = dequeToList(attributesDeque);
         relationships = dequeToList(relationshipsDeque);
+        entityPermissions = new EntityPermissions(cls, fieldOrMethodList);
     }
 
     /**
      * Bind fields of an entity including the Id field, attributes, and relationships.
      *
-     * @param cls Class type to bind fields
-     * @param type JSON API type identifier
+     * @param cls               Class type to bind fields
```

Comment:
Just a comment about whitespace like this: it actually is hard to maintain (due to variable name changes, etc.) and doesn't buy very much. I suggest preferring to simply use a single space rather than aligning.